[Xcode] Move WebKitLegacy's header migration logic to a script phase

emw-apple opened the pull request:
#### 961540b8e2eeb3788116bea395e2900cea7ebc6d
<pre>
[Xcode] Move WebKitLegacy&apos;s header migration logic to a script phase
<a href="https://bugs.webkit.org/show_bug.cgi?id=252715">https://bugs.webkit.org/show_bug.cgi?id=252715</a>

Reviewed by Alexey Proskuryakov.

Apply the same refactoring that was done for WebKit in 259916@main.
Move the migrate-headers.py script to WebKitLegacy and use it to copy
headers from WebCore.

Unlike WebKit, which reexports WebKitLegacy, WebKitLegacy doesn&apos;t
formally reexport WebCore, so extra work is needed to tell TAPI about
the migrated headers. Use -extra-private-header to pass them to tapi.
This list must be maintained, but since the set of migrated headers is
stable and verified by TAPI, it should be a low maintenance burden. We
can change migrate-headers.py to generate a filelist once
rdar://102197642 is done.

* Source/WebKit/WebKit.xcodeproj/project.pbxproj: Refer to
  migrate-headers.py from its new location in WebKitLegacy.
* Source/WebKitLegacy/MigratedHeaders-input.xcfilelist: Added.
* Source/WebKitLegacy/MigratedHeaders-output.xcfilelist: Added.
* Source/WebKitLegacy/WebKitLegacy.xcodeproj/project.pbxproj:
  - Remove &quot;Migrated Headers&quot; group.
  - Add migrate-headers.py script and the xcfilelists that power it.
  - Replace the migrate-header-rule build rule with a script phase that calls migrate-headers.py.
  - Update the &quot;Generate Export Files&quot; script phase to read a single
    `ReexportedFromWebCore.exp` file.
* Source/WebKitLegacy/ios/WebCoreSupport/WebInspectorClientIOS.mm: I&apos;m
  not exactly sure why, but the compiler now complains about some
  missing implementation methods in the stubbed-out
  WebInspectorFrontendClient for iOS. Add them.
(WebInspectorFrontendClient::supportsDiagnosticLogging):
(WebInspectorFrontendClient::logDiagnosticEvent):
(WebInspectorFrontendClient::canSave):
* Source/WebKitLegacy/mac/Configurations/WebKitLegacy.xcconfig: Add
  -extra-private-header flags and leave FIXMEs about when to get rid of
  it.
* Source/WebKitLegacy/scripts/migrate-header-rule:
  - Stop creating forwarding headers. Because the migrated headers
    aren&apos;t part of WebKitLegacy&apos;s target, they aren&apos;t part of its
    headermap, which means we no longer need to support parsing
    unprocessed WebCore headers with other &lt;WebCore/...&gt; includes.
  - Instead of creating one export file per header, write them all to a
    single `ReexportedFromWebCore.exp`.
* Source/WebKitLegacy/scripts/migrate-headers.py: Renamed from Source/WebKit/Scripts/migrate-headers.py.

Canonical link: <a href="https://commits.webkit.org/260718@main">https://commits.webkit.org/260718@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/97621542b7a5f4bb4b1f3fd820e43c5cb5a808b8

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/109160 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/18242 "Built successfully") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/43/builds/41976 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/690 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/118404 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/19701 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/9544 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/101400 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/114918 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/14774 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/43/builds/41976 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/42942 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/96746 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/43/builds/41976 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/84675 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/11021 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/43/builds/41976 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/82/builds/11750 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/7933 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/80/builds/17119 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/43/builds/41976 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7393 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/13368 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov commented on Source/WebKitLegacy/mac/Configurations/WebKitLegacy.xcconfig:
```diff
@@ -124,9 +122,17 @@ TAPI_USE_SRCROOT = $(TAPI_USE_SRCROOT$(WK_XCODE_15));
 TAPI_USE_SRCROOT_XCODE_SINCE_15 = YES;
 TAPI_VERIFY_MODE[sdk=iphone*] = Pedantic;
 
+// FIXME: Replace -extra-*-header and -exclude-*-header flags with build setting equivalents once rdar://99885053 is completed in all needed toolchains.
```
Is this the right radar? Looking at its title, I don't understand the relationship.

emw-apple committed:
[Xcode] Move WebKitLegacy's header migration logic to a script phase
https://bugs.webkit.org/show_bug.cgi?id=252715

Reviewed by Alexey Proskuryakov.

Apply the same refactoring that was done for WebKit in 259916@main.
Move the migrate-headers.py script to WebKitLegacy and use it to copy
headers from WebCore.

Unlike WebKit, which reexports WebKitLegacy, WebKitLegacy doesn't
formally reexport WebCore, so extra work is needed to tell TAPI about
the migrated headers. Use -extra-private-header to pass them to tapi.
This list must be maintained, but since the set of migrated headers is
stable and verified by TAPI, it should be a low maintenance burden. We
can change migrate-headers.py to generate a filelist once
rdar://102197642 is done.

* Source/WebKit/WebKit.xcodeproj/project.pbxproj: Refer to
  migrate-headers.py from its new location in WebKitLegacy.
* Source/WebKitLegacy/MigratedHeaders-input.xcfilelist: Added.
* Source/WebKitLegacy/MigratedHeaders-output.xcfilelist: Added.
* Source/WebKitLegacy/WebKitLegacy.xcodeproj/project.pbxproj:
  - Remove "Migrated Headers" group.
  - Add migrate-headers.py script and the xcfilelists that power it.
  - Replace the migrate-header-rule build rule with a script phase that calls migrate-headers.py.
  - Update the "Generate Export Files" script phase to read a single
    `ReexportedFromWebCore.exp` file.
* Source/WebKitLegacy/ios/WebCoreSupport/WebInspectorClientIOS.mm: I'm
  not exactly sure why, but the compiler now complains about some
  missing implementation methods in the stubbed-out
  WebInspectorFrontendClient for iOS. Add them.
(WebInspectorFrontendClient::supportsDiagnosticLogging):
(WebInspectorFrontendClient::logDiagnosticEvent):
(WebInspectorFrontendClient::canSave):
* Source/WebKitLegacy/mac/Configurations/WebKitLegacy.xcconfig: Add
  -extra-private-header flags and leave FIXMEs about when to get rid of
  it.
* Source/WebKitLegacy/scripts/migrate-header-rule:
  - Stop creating forwarding headers. Because the migrated headers
    aren't part of WebKitLegacy's target, they aren't part of its
    headermap, which means we no longer need to support parsing
    unprocessed WebCore headers with other <WebCore/...> includes.
  - Instead of creating one export file per header, write them all to a
    single `ReexportedFromWebCore.exp`.
* Source/WebKitLegacy/scripts/migrate-headers.py: Renamed from Source/WebKit/Scripts/migrate-headers.py.

Canonical link: https://commits.webkit.org/260718@main
